app/models/live_metric.rb

```diff
@@ -0,0 +1,112 @@
+class LiveMetric < ActsAsArModel
+  set_columns_hash(:timestamp => :datetime)
+
+  class LiveMetricError < RuntimeError; end
+
+  def self.where(*)
+    raise LiveMetricError, "where method is not supported on LiveMetric"
+  end
+
+  def self.find(*args)
+    validate_args(args)
+    raw_query = args[1]
+    processed = process_conditions(raw_query[:conditions])
+    resource = fetch_resource(processed[:resource_type], processed[:resource_id])
+    if resource.nil?
+      []
+    else
+      filter_and_fetch_metrics(resource, raw_query[:ext_options][:only_cols], processed[:start_time],
+                               processed[:end_time], processed[:interval_name])
+    end
+  end
+
+  def self.validate_args(args)
+    unless args && args.length > 1
+      raise LiveMetricError, "LiveMetrics expression wrong number of arguments"
```

Comment:
Do you suggest to change the signature to something like def self.validate_args(arg0, arg1) ?
And invoking the method with validate_args(args[0], args[1]) ?
Probably it's me but I don't see it. Could you elaborate the example ?
(Disclaimer: I'm still thinking with Java patterns in mind, so probably I don't use Ruby best practices correctly)